incubator/hnc/internal/reconcilers/object.go

```diff
@@ -793,6 +804,26 @@ func (r *ObjectReconciler) shouldPropagateSource(log logr.Logger, inst *unstruct
 	}
 }
 
+// getSelector returns the selector on a given object if it exists
+func (r *ObjectReconciler) getSelector(log logr.Logger, inst *unstructured.Unstructured) labels.Selector {
+	annot := inst.GetAnnotations()
+	selectorStr, ok := annot[api.AnnotationSelector]
+	if !ok {
+		return nil
+	}
+	labelSelector, err := v1.ParseToLabelSelector(selectorStr)
+	// TODO: surface the error messages here
```

Comment:
File issue?

Reply:
Also FYI I was writing a "HNC Exceptions - handling errors" doc on this with possible events or webhooks.

Reply:
done